### ISSUE TITLE
Error: contract runner does not support sending transactions (operation="sendTransaction", code=UNSUPPORTED_OPERATION, version=6.7.1)  (URGENT)

### ISSUE THREAD

**Issue by yassineAmmar** [adrianhajdin/project_web3.0#112]
in the video at 2:07:39 when i click on send now  then metaMask popup and i click confirm just after that i get the error 
this is my code 
// this is my getEthereumContract function
const getEthereumContract = ()=> {
    const provider = new ethers.BrowserProvider(window.ethereum);
    const signer = provider.getSigner()
    const TransactionContract = new ethers.Contract(contractAddresse, contractABI, signer);

    return TransactionContract
}

// this is my sendTransaction function

    const sendTransaction = async () => {
        try {
            if (!ethereum) return alert("Please install MetaMask.");

            // Ensure that currentAccount is set correctly chatgpt
            if (!currentAccount) {
                alert("Please connect your wallet.");
                return;
            }

            const {addressTo, amount, keyword, message} = formData
            const TransactionContract =  getEthereumContract()
            // const parsedAmount = ethers.utils.parseEther(amount); //convert the decimal amount to gwie 
            const parsedAmount = amount * 10e17;

            await ethereum.request({
                method : 'eth_sendTransaction',
                params: [{
                    from: currentAccount,
                    to: addressTo,
                    gas:'0x5208', //21000 gwie
                    value: parsedAmount._hex,
                }]
            })

            // store the transaction in the blockChain using function from Transactions.sol
            const transactionHash = await TransactionContract.addToBlockchain(addressTo, parsedAmount, keyword, message)

            setIsLoading(true)            
            console.log(`loading - ${transactionHash.hash}`);
            await transactionHash.wait()
            setIsLoading(false)            
            console.log(`success ${transactionHash.hash}`);
            const transactionCount = await TransactionContract.getTransactionsCount()
            setTransactionCount(transactionCount.toNumber())

        } catch (error) {
            console.log(error);
            throw new Error("No ethereum object");
        }
    }
![Screenshot from 2023-09-01 13-04-16](https://github.com/adrianhajdin/project_web3.0/assets/126362301/59a07c61-0919-4a96-8ec8-cd011ccbcc9f)

**Comment by William2809** [comment]
@yassineAmmar how did you solve the Error: contract runner does not support sending transactions (operation="sendTransaction", code=UNSUPPORTED_OPERATION, version=6.7.1) ? i stucked in this error

**Comment by William2809** [comment]
Nvm, i solved it by adding await in the getEthereumContract:

`const getEthereumContract = async () => {
	const provider = new ethers.BrowserProvider(ethereum);
	const signer = await provider.getSigner();
	const transactionContract = new ethers.Contract(
		contractAddress,
		contractABI,
		signer
	);

	return transactionContract;
};`

**Comment by yassineAmmar** [comment]
sorry mate i've just seen your message , i rarely check my mail, so i guess
you learning blockchain too if you like we could stay on contact to help
each other on this journey



Le mer. 20 sept. 2023 à 16:51, William2809 ***@***.***> a
écrit :

**Comment by William2809** [comment]
Yes, i just started learning last week. It is really tricky because the library got updated and lots of things were changed. Sure, we could exchange contact, it would be really helpful for me too 😁

**Comment by yassineAmmar** [comment]
+212 659-291933
That's my Whatsapp please text me

**Comment by RushabhM03** [comment]
@yassineAmmar how did you solve the issue where no error was displayed but no ether was getting transferred to the other account?
